website_event_filter_selector/controllers/main.py

```diff
@@ -1,110 +1,59 @@
-        # separate method and make our life easier, but not happening now.
-        domain = [("state", "in", ("draft", "confirm", "done"))]
-
         def dom_without(without):
-            return [leaf for leaf in domain if leaf[0] not in without]
-
-        # Date domain
-        for date in values["dates"]:
-            if values["searches"]["date"] == date[0]:
-                domain += date[2]
-                break
-
-        # Type domain
-        if values["current_type"]:
-            domain.append(("event_type_id", "=", values["current_type"].id))
-
-        # Country domain
-        if values["current_country"]:
-            domain.append(("country_id", "=", values["current_country"].id))
-        elif searches["country"] == "online":
-            domain.append(("country_id", "=", False))
+            domain = [("state", "in", ["draft", "confirm", "done"])]
+            for key, search in domain_search.items():
+                if key != without:
+                    domain += search
+            return domain
 
         # Handle city search
-        Event = http.request.env["event.event"].with_context(http.request.context)
+        domain = dom_without("city")
         cities = Event.read_group(domain, ["city"], groupby="city", orderby="city")
         cities.insert(
             0,
             {
                 "city_count": sum(x["city_count"] for x in cities),
-                "city": _("All Cities"),
-                "key": "all",
+                "city": ("All Cities"),
             },
         )
         values["cities"] = cities
-        values["current_city"] = searches["city"]
+        searches.setdefault("city", cities[0]["city"])
 
-        if searches["city"] != "all":
+        current_city = None
+        if searches["city"] != cities[0]["city"]:
             domain.append(("city", "=", searches["city"]))
-            # Patch type count
-            values["types"][1:] = Event.read_group(
-                dom_without({"event_type_id"}),
-                ["id", "event_type_id"],
-                groupby=["event_type_id"],
-                orderby="event_type_id",
-            )
-            values["types"][0]["event_type_id_count"] = sum(
-                int(type_["event_type_id_count"]) for type_ in values["types"][1:]
-            )
-            # Patch country count
-            values["countries"][1:] = Event.read_group(
-                dom_without({"country_id"}),
-                ["id", "country_id"],
-                groupby="country_id",
-                orderby="country_id",
-            )
-            values["countries"][0]["country_id_count"] = sum(
-                int(type_["country_id_count"]) for type_ in values["countries"][1:]
-            )
-            # Patch date count
-            for date in values["dates"]:
-                if date[0] != "old":
-                    date[3] = Event.search_count(
-                        dom_without({"date_end", "date_begin"}) + date[2]
-                    )
```

Comment:
I think that, by removing these lines, you're introducing a bug.

At least in previous versions, when you filtered by one criteria, other criterias' counters were updated accordingly.

Example:

1. I have 2 Trainings in Spain and 2 trainings in France.
2. "All types" displays count 4️⃣ 
3. I filter by country=France
4. "All types" displays count 2️⃣ 

The same should apply to the cities filter here.